src/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilder.java

```diff
@@ -445,16 +445,25 @@ else if (STREAM_TYPES.contains(this.outType))
         return writer;
      }
 
-    private void determineOutputTypeFromFilename() {
-        if (isBCF(this.outFile)) {
+    private void determineOutputTypeFromFilename(final File f) {
+        if (isBCF(f)) {
             this.outType = OutputType.BCF;
-        } else if (isCompressedVCF(this.outFile)) {
+        } else if (isCompressedVCF(f)) {
             this.outType = OutputType.BLOCK_COMPRESSED_VCF;
-        } else if (isVCF(this.outFile)) {
+        } else if (isVCF(f)) {
             this.outType = OutputType.VCF;
         }
         else {
-            this.outType = OutputType.UNSPECIFIED;
+            // See if we have a special file (device, named pipe, etc.)
+            final File canonical = new File(IOUtil.getFullCanonicalPath(this.outFile));
+            if (!canonical.equals(f)) {
```

Comment:
Can you document in *precisely* which cases this would be true (`canonical` not equal to `f`)? I understand that resolving a symlink could cause this, obviously, but could you list/explain any other cases you had in mind?

Also, am I wrong in thinking that a symlink to a `vcf` (for example) would typically be caught by the `isVCF()` check above, assuming the symlink itself ends in `.vcf`?